share/smack/lib/smack.c

```diff
@@ -982,6 +982,7 @@ void __SMACK_decls() {
 
 #else // NO_REUSE does not reuse previously-allocated addresses
   D("var $Alloc: [ref] bool;");
+  D("function $Size(ref) returns (ref);");
```

Comment:
Wait, you did not add `$Size` to `NO_REUSE_IMPLS`. That is our main used memory model, and so please do that. You will have to introduce the appropriate constraints on base there too!